src/modules/gallery/containers/photo-upload-modal/photo-upload-modal.tsx

```diff
@@ -0,0 +1,336 @@
+// components
+import { Dialog } from '@/modules/core/components/dialog';
+import { Typography } from '@/modules/core/components/typogrpahy';
+import { Icon } from '@/modules/core/components/icon';
+import { Button } from '@/modules/core/components/button';
+import { Avatar } from '@/modules/core/components/avatar';
+import { TextField } from '@/modules/core/components/text-field';
+import { Placeholder } from '@/modules/core/components/placeholder';
+// hooks
+import { useImageUploadMutation } from '@/modules/user/hooks/use-image-upload-mutation';
+// utils
+import { getFullName } from '@/modules/user/utils/user.utils';
+import { trpc } from '@/modules/core/utils/trpc.utils';
+import { showToast } from '@/modules/core/providers/toast-provider';
+import { dataURLtoBlob } from '@/modules/core/utils/file.utils';
+
+import 'react-advanced-cropper/dist/style.css';
+import 'react-advanced-cropper/dist/themes/bubble.css';
+
+import { PhotoUploadModalHeader } from './elements/photo-upload-modal-header';
+
+import type {
+  PhotoUploadState,
+  PhotoUploadModalProps,
+} from './photo-upload-modal.interface';
+import { Spinner } from '@/modules/core/components/spinner';
+
+export const PhotoUploadModal: FC<PhotoUploadModalProps> = ({
+  isOpen,
+  onOpenChange,
+  trigger,
+  portfolioId,
+}) => {
+  const queryClient = useQueryClient();
+  // queries
+  const { data: me } = trpc.user.me.useQuery({
+    expand: ['professional'],
+  });
+  const uploadedPortfolio = trpc.portfolio.get.useQuery(
+    {
+      id: portfolioId || '',
+    },
+    {
+      enabled: !!portfolioId && isOpen,
+    }
+  );
+  // animation
+  const springs = useSpringValue(0);
+  // state
+  const [state, setState] = useState<PhotoUploadState>({
+    step: portfolioId ? 'preview' : 'select',
+  });
+  const [description, setDescription] = useState('');
+  // refs
+  const cropperRef = useRef<CropperRef>(null);
+  // mutations
+  const uploadMutation = useImageUploadMutation();
+  const portfolioCreateMutation = trpc.portfolio.create.useMutation();
+  const portfolioUpdateMutation = trpc.portfolio.update.useMutation();
+
+  const { getRootProps, getInputProps, isDragAccept } = useDropzone({
+    disabled: state.step !== 'select',
+    accept: {
+      'image/jpeg': [],
+      'image/jpg': [],
+      'image/png': [],
+    },
+    maxFiles: 1,
+    onDrop: (acceptedFilesDropped) => {
+      if (acceptedFilesDropped.length) {
+        setState({
+          step: 'crop',
+          file: acceptedFilesDropped[0],
+          preview: URL.createObjectURL(acceptedFilesDropped[0]),
+        });
+      }
+    },
+  });
+
+  useEffect(() => {
+    if (isOpen) {
+      setState({ step: portfolioId ? 'preview' : 'select' });
+    }
+  }, [isOpen, portfolioId]);
+
+  useEffect(() => {
+    if (portfolioId && uploadedPortfolio.data) {
+      setDescription(uploadedPortfolio.data.description || '');
+    }
+  }, [uploadedPortfolio.data, portfolioId]);
```

Comment:
We need state in this case since `description` field is editable in the modal. This state is connected to textarea as well via value and onChange. The `useEffect` is used to set initial value.